source/Nevermore.IntegrationTests/RelationalTransaction/LoadFixture.cs

```diff
@@ -240,5 +252,70 @@ public void StoreAndLoadStringInheritedTypesThatAreProperties()
                 allMachines.Single(x => x.Name == "Machine A").Endpoint.Should().BeOfType<PassiveTentacleEndpoint>();
             }
         }
+
+        [Test]
+        public void StoreAndLoadFromParameterizedRawSql()
+        {
+            using (var trn = Store.BeginTransaction())
+            {
+                InsertProductAndLineItems("Unicorn Hair", 2m, trn, 1);         // subtotal: $2 of Unicorn Hair
+                InsertProductAndLineItems("Unicorn Poop", 3m, trn, 2, 3);      // subtotal: $15 of Unicorn Poop
+                InsertProductAndLineItems("Unicorn Dust", 1m, trn, 2, 1, 7);   // subtotal: $10 of Unicorn Dust
+                InsertProductAndLineItems("Fairy Bread", 10m, trn, 4);         // subtotal: $40 of Fairy Bread
+                trn.Commit();
+
+                var productSubtotalQuery =    @"SELECT
+                                                    Id,
+                                                    ProductId,
+                                                    ProductName,
+                                                    Subtotal
+                                                FROM (
+                                                    SELECT 
+                                                        CAST(NEWID() AS nvarchar(36)) Id,
+                                                        p.ProductId,
+                                                        p.ProductName,
+                                                        SUM(p.Price * l.Quantity) Subtotal
+                                                    FROM (
+                                                        SELECT
+                                                            ProductId,
+                                                            CAST(JSON_VALUE([JSON], '$.Quantity') AS int) Quantity
+                                                        FROM LineItem
+                                                    ) l
+                                                    INNER JOIN (
+                                                    SELECT 
+                                                        Id ProductId,
+                                                        [Name] ProductName,
+                                                        CAST(JSON_VALUE([JSON], '$.Price') AS decimal) Price
+                                                        FROM Product
+                                                    ) p ON p.ProductId = l.ProductId
+                                                    GROUP BY p.ProductId, p.ProductName
+                                                ) p3
+                                                WHERE Subtotal >= @MinimumSubtotal";
+
+                var doubleDigitUnicornProductSubtotals = trn.RawSqlQuery<ProductSubtotal>(productSubtotalQuery)
+                    .Where(s => s.ProductName.Contains("Unicorn"))
```

Comment:
That's an interesting one. How does it know to add an `AND` instead of `WHERE`

Reply:
The raw SQL is wrapped up as a subquery, so the outer `Where` applies to that:
```
SELECT *
FROM (
    ...
    WHERE Subtotal >= @MinimumSubtotal
) ALIAS_GENERATED_1
WHERE ([ProductName] LIKE @productname_0)
```